queue_consumer.go

```diff
@@ -18,17 +18,18 @@ var (
 // NewConsumer creates a Consumer that uses the given SQSService to connect and invokes the handler for each message received.
 func NewConsumer(s *SQSService, handler MessageHandlerFunc) *Consumer {
 	return &Consumer{
-		s:                              s,
-		handler:                        handler,
-		delayAfterReceiveError:         defaultDelayAfterReceiveError,
-		WaitSeconds:                    defaultReceiveMessageWaitSeconds,
-		ReceiveVisibilityTimoutSeconds: defaultReceiveVisibilityTimeoutSeconds,
-		Logger:                         NoopLogger,
-
+		s:                                s,
+		handler:                          handler,
+		Logger:                           NoopLogger,
+		JobWorkersCount:                  defaultJobWorkersCount,
```

Comment:
Here we don't pass the custom value, but use the default value. So I guess we will set the custom value after a new consumer is created, like this: `q.JobWorkersCount = 5` ?

Reply:
yes, we can pass this config as opt struct `NewConsumer(service, handler, opts)` but that will break services that call NewConsumer() but ideally, should be passed as `opt` struct instead

Reply:
You could add variable length argument like we have in the `Run()` method: https://github.com//Wattpad/sqsconsumer/blob/2a30a1f44ed55c495154708a7dfc132f3fa4085a/queue_consumer.go#L140-L141

This would allow new callers to add optional config settings without breaking existing callers.

Reply:
However, now that I look at it, it seems this package has already established a pattern of exposing fields and letting you change them.  I guess there isn't really a need to change that.